lib/beaker/dsl/helpers.rb

```diff
@@ -1355,6 +1355,43 @@ def curl_on(host, cmd, opts = {}, &block)
           on host, "curl %s" % cmd, opts, &block
         end
       end
+
+      # Write hiera config file on one or more provided hosts
+      #
+      # @param[Host, Array<Host>, String, Symbol] host    One or more hosts to act upon,
+      #                           or a role (String or Symbol) that identifies one or more hosts.
+      # @param[Array] One or more hierarchy paths
+      def write_hiera_config_on(host, hierarchy)
+        hiera_config=Hash.new
+        hiera_config[:backends] = 'yaml'
+        hiera_config[:yaml] = {}
+        hiera_config[:yaml][:datadir] = host[:hieradatadir]
```

Comment:
This won't work on multi-host.  You need to use the block_on method to be able to cycle through all potential hosts passed to this.